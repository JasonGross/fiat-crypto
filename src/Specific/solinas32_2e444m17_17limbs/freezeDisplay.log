freeze = λ var : Z.Syntax.base_type → Type,
λ '(x30, x31, x29, x27, x25, x23, x21, x19, x17, x15, x13, x11, x9, x7, x5, x3, x1)%core,
uint32_t x33, uint32_t x34 = Op (Z.Syntax.SubWithGetBorrow 27 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (0x0, Return x1, 0x7ffffef);
uint32_t x36, uint32_t x37 = Op (Z.Syntax.SubWithGetBorrow 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x34, Return x3, 0x3ffffff);
uint32_t x39, uint32_t x40 = Op (Z.Syntax.SubWithGetBorrow 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x37, Return x5, 0x3ffffff);
uint32_t x42, uint32_t x43 = Op (Z.Syntax.SubWithGetBorrow 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x40, Return x7, 0x3ffffff);
uint32_t x45, uint32_t x46 = Op (Z.Syntax.SubWithGetBorrow 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x43, Return x9, 0x3ffffff);
uint32_t x48, uint32_t x49 = Op (Z.Syntax.SubWithGetBorrow 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x46, Return x11, 0x3ffffff);
uint32_t x51, uint32_t x52 = Op (Z.Syntax.SubWithGetBorrow 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x49, Return x13, 0x3ffffff);
uint32_t x54, uint32_t x55 = Op (Z.Syntax.SubWithGetBorrow 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x52, Return x15, 0x3ffffff);
uint32_t x57, uint32_t x58 = Op (Z.Syntax.SubWithGetBorrow 27 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x55, Return x17, 0x7ffffff);
uint32_t x60, uint32_t x61 = Op (Z.Syntax.SubWithGetBorrow 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x58, Return x19, 0x3ffffff);
uint32_t x63, uint32_t x64 = Op (Z.Syntax.SubWithGetBorrow 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x61, Return x21, 0x3ffffff);
uint32_t x66, uint32_t x67 = Op (Z.Syntax.SubWithGetBorrow 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x64, Return x23, 0x3ffffff);
uint32_t x69, uint32_t x70 = Op (Z.Syntax.SubWithGetBorrow 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x67, Return x25, 0x3ffffff);
uint32_t x72, uint32_t x73 = Op (Z.Syntax.SubWithGetBorrow 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x70, Return x27, 0x3ffffff);
uint32_t x75, uint32_t x76 = Op (Z.Syntax.SubWithGetBorrow 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x73, Return x29, 0x3ffffff);
uint32_t x78, uint32_t x79 = Op (Z.Syntax.SubWithGetBorrow 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x76, Return x31, 0x3ffffff);
uint32_t x81, uint32_t x82 = Op (Z.Syntax.SubWithGetBorrow 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x79, Return x30, 0x3ffffff);
uint32_t x83 = cmovznz32(x82, 0x0, 0xffffffff);
uint32_t x84 = (x83 & 0x7ffffef);
uint32_t x86, uint32_t x87 = Op (Z.Syntax.AddWithGetCarry 27 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (0x0, Return x33, Return x84);
uint32_t x88 = (x83 & 0x3ffffff);
uint32_t x90, uint32_t x91 = Op (Z.Syntax.AddWithGetCarry 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x87, Return x36, Return x88);
uint32_t x92 = (x83 & 0x3ffffff);
uint32_t x94, uint32_t x95 = Op (Z.Syntax.AddWithGetCarry 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x91, Return x39, Return x92);
uint32_t x96 = (x83 & 0x3ffffff);
uint32_t x98, uint32_t x99 = Op (Z.Syntax.AddWithGetCarry 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x95, Return x42, Return x96);
uint32_t x100 = (x83 & 0x3ffffff);
uint32_t x102, uint32_t x103 = Op (Z.Syntax.AddWithGetCarry 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x99, Return x45, Return x100);
uint32_t x104 = (x83 & 0x3ffffff);
uint32_t x106, uint32_t x107 = Op (Z.Syntax.AddWithGetCarry 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x103, Return x48, Return x104);
uint32_t x108 = (x83 & 0x3ffffff);
uint32_t x110, uint32_t x111 = Op (Z.Syntax.AddWithGetCarry 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x107, Return x51, Return x108);
uint32_t x112 = (x83 & 0x3ffffff);
uint32_t x114, uint32_t x115 = Op (Z.Syntax.AddWithGetCarry 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x111, Return x54, Return x112);
uint32_t x116 = (x83 & 0x7ffffff);
uint32_t x118, uint32_t x119 = Op (Z.Syntax.AddWithGetCarry 27 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x115, Return x57, Return x116);
uint32_t x120 = (x83 & 0x3ffffff);
uint32_t x122, uint32_t x123 = Op (Z.Syntax.AddWithGetCarry 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x119, Return x60, Return x120);
uint32_t x124 = (x83 & 0x3ffffff);
uint32_t x126, uint32_t x127 = Op (Z.Syntax.AddWithGetCarry 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x123, Return x63, Return x124);
uint32_t x128 = (x83 & 0x3ffffff);
uint32_t x130, uint32_t x131 = Op (Z.Syntax.AddWithGetCarry 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x127, Return x66, Return x128);
uint32_t x132 = (x83 & 0x3ffffff);
uint32_t x134, uint32_t x135 = Op (Z.Syntax.AddWithGetCarry 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x131, Return x69, Return x132);
uint32_t x136 = (x83 & 0x3ffffff);
uint32_t x138, uint32_t x139 = Op (Z.Syntax.AddWithGetCarry 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x135, Return x72, Return x136);
uint32_t x140 = (x83 & 0x3ffffff);
uint32_t x142, uint32_t x143 = Op (Z.Syntax.AddWithGetCarry 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x139, Return x75, Return x140);
uint32_t x144 = (x83 & 0x3ffffff);
uint32_t x146, uint32_t x147 = Op (Z.Syntax.AddWithGetCarry 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x143, Return x78, Return x144);
uint32_t x148 = (x83 & 0x3ffffff);
uint32_t x150, uint32_t _ = Op (Z.Syntax.AddWithGetCarry 26 (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5)) (Return x147, Return x81, Return x148);
(Return x150, Return x146, Return x142, Return x138, Return x134, Return x130, Return x126, Return x122, Return x118, Return x114, Return x110, Return x106, Return x102, Return x98, Return x94, Return x90, Return x86)
     : ∀ var : Z.Syntax.base_type → Type, expr Z.Syntax.base_type Z.Syntax.op (uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t -> uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t)

Argument scope is [function_scope]
