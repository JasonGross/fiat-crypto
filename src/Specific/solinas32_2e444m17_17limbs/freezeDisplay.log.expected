λ x : word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32,
Interp-η
(λ var : Syntax.base_type → Type,
 λ '(x31, x32, x30, x28, x26, x24, x22, x20, x18, x16, x14, x12, x10, x8, x6, x4, x2)%core,
 uint32_t x34, uint8_t/*bool*/ x35 = Op (Syntax.SubWithGetBorrow 27 (Syntax.TWord 0) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 0)) (0x0, Return x2, 0x7ffffef);
 uint32_t x37, uint8_t/*bool*/ x38 = subborrow_u26(x35, x4, 0x3ffffff);
 uint32_t x40, uint8_t/*bool*/ x41 = subborrow_u26(x38, x6, 0x3ffffff);
 uint32_t x43, uint8_t/*bool*/ x44 = subborrow_u26(x41, x8, 0x3ffffff);
 uint32_t x46, uint8_t/*bool*/ x47 = subborrow_u26(x44, x10, 0x3ffffff);
 uint32_t x49, uint8_t/*bool*/ x50 = subborrow_u26(x47, x12, 0x3ffffff);
 uint32_t x52, uint8_t/*bool*/ x53 = subborrow_u26(x50, x14, 0x3ffffff);
 uint32_t x55, uint8_t/*bool*/ x56 = subborrow_u26(x53, x16, 0x3ffffff);
 uint32_t x58, uint8_t/*bool*/ x59 = Op (Syntax.SubWithGetBorrow 27 (Syntax.TWord 0) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 0)) (Return x56, Return x18, 0x7ffffff);
 uint32_t x61, uint8_t/*bool*/ x62 = subborrow_u26(x59, x20, 0x3ffffff);
 uint32_t x64, uint8_t/*bool*/ x65 = subborrow_u26(x62, x22, 0x3ffffff);
 uint32_t x67, uint8_t/*bool*/ x68 = subborrow_u26(x65, x24, 0x3ffffff);
 uint32_t x70, uint8_t/*bool*/ x71 = subborrow_u26(x68, x26, 0x3ffffff);
 uint32_t x73, uint8_t/*bool*/ x74 = subborrow_u26(x71, x28, 0x3ffffff);
 uint32_t x76, uint8_t/*bool*/ x77 = subborrow_u26(x74, x30, 0x3ffffff);
 uint32_t x79, uint8_t/*bool*/ x80 = subborrow_u26(x77, x32, 0x3ffffff);
 uint32_t x82, uint8_t/*bool*/ x83 = subborrow_u26(x80, x31, 0x3ffffff);
 uint32_t x84 = cmovznz32(x83, 0x0, 0xffffffff);
 uint32_t x85 = (x84 & 0x7ffffef);
 uint32_t x87, uint8_t/*bool*/ x88 = Op (Syntax.AddWithGetCarry 27 (Syntax.TWord 0) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 0)) (0x0, Return x34, Return x85);
 uint32_t x89 = (x84 & 0x3ffffff);
 uint32_t x91, uint8_t/*bool*/ x92 = addcarryx_u26(x88, x37, x89);
 uint32_t x93 = (x84 & 0x3ffffff);
 uint32_t x95, uint8_t/*bool*/ x96 = addcarryx_u26(x92, x40, x93);
 uint32_t x97 = (x84 & 0x3ffffff);
 uint32_t x99, uint8_t/*bool*/ x100 = addcarryx_u26(x96, x43, x97);
 uint32_t x101 = (x84 & 0x3ffffff);
 uint32_t x103, uint8_t/*bool*/ x104 = addcarryx_u26(x100, x46, x101);
 uint32_t x105 = (x84 & 0x3ffffff);
 uint32_t x107, uint8_t/*bool*/ x108 = addcarryx_u26(x104, x49, x105);
 uint32_t x109 = (x84 & 0x3ffffff);
 uint32_t x111, uint8_t/*bool*/ x112 = addcarryx_u26(x108, x52, x109);
 uint32_t x113 = (x84 & 0x3ffffff);
 uint32_t x115, uint8_t/*bool*/ x116 = addcarryx_u26(x112, x55, x113);
 uint32_t x117 = (x84 & 0x7ffffff);
 uint32_t x119, uint8_t/*bool*/ x120 = Op (Syntax.AddWithGetCarry 27 (Syntax.TWord 0) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 5) (Syntax.TWord 0)) (Return x116, Return x58, Return x117);
 uint32_t x121 = (x84 & 0x3ffffff);
 uint32_t x123, uint8_t/*bool*/ x124 = addcarryx_u26(x120, x61, x121);
 uint32_t x125 = (x84 & 0x3ffffff);
 uint32_t x127, uint8_t/*bool*/ x128 = addcarryx_u26(x124, x64, x125);
 uint32_t x129 = (x84 & 0x3ffffff);
 uint32_t x131, uint8_t/*bool*/ x132 = addcarryx_u26(x128, x67, x129);
 uint32_t x133 = (x84 & 0x3ffffff);
 uint32_t x135, uint8_t/*bool*/ x136 = addcarryx_u26(x132, x70, x133);
 uint32_t x137 = (x84 & 0x3ffffff);
 uint32_t x139, uint8_t/*bool*/ x140 = addcarryx_u26(x136, x73, x137);
 uint32_t x141 = (x84 & 0x3ffffff);
 uint32_t x143, uint8_t/*bool*/ x144 = addcarryx_u26(x140, x76, x141);
 uint32_t x145 = (x84 & 0x3ffffff);
 uint32_t x147, uint8_t/*bool*/ x148 = addcarryx_u26(x144, x79, x145);
 uint32_t x149 = (x84 & 0x3ffffff);
 uint32_t x151, uint8_t/*bool*/ _ = addcarryx_u26(x148, x82, x149);
 (Return x151, Return x147, Return x143, Return x139, Return x135, Return x131, Return x127, Return x123, Return x119, Return x115, Return x111, Return x107, Return x103, Return x99, Return x95, Return x91, Return x87))
x
     : word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 → ReturnType (uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t)
